app/Repositories/OfferRepository.php

```diff
@@ -12,4 +12,19 @@ public function __construct()
     {
         $this->model = new Offer();
     }
+
+    public function getNewOffers($userId, $categoryId = null)
+    {
+        $conditions = [
+            ["user_id", "!=", $userId],
+            ["status.code", "=", "new"]
+        ];
+        if ($categoryId) {
+            $conditions[] = ["category_id", "=", $categoryId];
+        }
+        return $this->model->select("start_at", "end_at", "address", "price_per_hour", "offer_categories.name as category_name")
```

Comment:
@RanaEmad I created a JSON resource class for retrieving offers with its related data in the "view_my_offers" branch, please take a look at it I think it will be better to use JSON resources instead. What do you think?

Reply:
Great! But this serves the same purpose in a different way of implementation and was created before you started working on your branch so can we just leave it as is if it's not breaking anything?